pkg/model/wasm_task.go

```diff
@@ -19,25 +19,24 @@ var _ JobType = (*WasmInputs)(nil)
 
 // UnmarshalInto implements taskUnmarshal
 func (wasm *WasmInputs) UnmarshalInto(with string, spec *Spec) error {
-	spec.Engine = EngineWasm
-	spec.Wasm = JobSpecWasm{
-		EntryPoint:           wasm.Entrypoint,
-		Parameters:           wasm.Parameters,
-		EnvironmentVariables: wasm.Env.Values,
-		ImportModules:        []StorageSpec{},
-	}
 
 	entryModule, err := parseResource(with)
 	if err != nil {
 		return err
 	}
 	spec.Inputs = []StorageSpec{parseStorageSource("/job", entryModule)}
 
+	var importModules []StorageSpec
 	for _, resource := range wasm.Modules {
+		// TODO(forrest): [correctness] this is correct but easily bug prone: https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable
 		resource := resource
-		spec.Wasm.ImportModules = append(spec.Wasm.ImportModules, parseStorageSource("", &resource))
+		importModules = append(importModules, parseStorageSource("", &resource))
 	}
 
+	// TODO(forrest): [logic change] I am not sure this is the correct way to set an entryModule, previously we never set it, attempting to now.
+	spec.EngineSpec = NewWasmEngineSpec(parseStorageSource("", entryModule), wasm.Entrypoint, wasm.Parameters, wasm.Env.Values, importModules)
+	spec.EngineDeprecated = EngineWasm
```

Comment:
This needs review:
previously we never assigned the wasm engine an entry module and instead put it in the Inputs, I have keep the behavior the same regarding inputs, but have changed the logic slightly by also including it in the wasm engine, I am not sure what the original intention of this method is.